src/variable.jl

```diff
@@ -310,82 +310,61 @@ nomissing(a::AbstractArray,value) = a
 export nomissing
 
 
-function Base.getindex(v::Variable,indexes::Int...)
+function readblock!(v::Variable, aout, indexes::Int...)
     datamode(v.ds)
-    return nc_get_var1(eltype(v),v.ds.ncid,v.varid,[i-1 for i in indexes[ndims(v):-1:1]])
+    aout .= nc_get_var1(eltype(v),v.ds.ncid,v.varid,[i-1 for i in indexes[ndims(v):-1:1]])
 end
 
-function Base.setindex!(v::Variable{T,N},data,indexes::Int...) where N where T
-    @debug "$(@__LINE__)"
+function readblock!(v::Variable{T,N}, aout, indexes::TR...) where {T,N} where TR <: Union{StepRange{Int,Int},UnitRange{Int}}
+    start,count,stride,jlshape = ncsub(indexes[1:N])
```

Comment:
```suggestion
    start,count,stride,jlshape = ncsub(indexes)
```

This is all I meant to change